Plugins/Public/base_plugin/PlayerCommands.cpp

```diff
@@ -1825,6 +1828,54 @@ namespace PlayerCommands
 		}
 	}
 
+	void GetNecessitiesStatus(uint client, const wstring &args) {
+		// Check that this player is in a player controlled base
+		PlayerBase *base = GetPlayerBaseForClient(client);
+		if (!base)
+		{
+			PrintUserCmdText(client, L"ERR Not in player base");
+			return;
+		}
+
+		if (!clients[client].admin && !clients[client].viewshop)
+		{
+			PrintUserCmdText(client, L"ERR Access denied");
+			return;
+		}
+
+        uint crewItemCount = base->HasMarketItem(set_base_crew_type);

```

Comment:
Something weird about the indentation here and on the two lines below. Mix of spaces and tabs?